.github/workflows/ci.yml

```diff
@@ -0,0 +1,69 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: tests
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Help Codecov detect commit SHA
+          fetch-depth: 2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 15
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      - name: 🧪 Run Tests
+        run: yarn test -- --single-run
+      - name: Codecov
+        run: bash <(curl -s https://codecov.io/bash)
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 15
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      - name: 💨 ESLint
+        run: yarn lint
+  typescript:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 15
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      - name: ✅ TypeScript lint
+        run: yarn lint:ts
+  prettier:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 15
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      - name: 💅 Prettier Check
+        run: yarn prettier:check
+  yarn-check-cache:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 15
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache --check-cache
```

Comment:
I think this is sorta a Yarn v2/Yarn PnP thing since the cache is in the repo instead of being pulled from npm/cached on the CI. But maybe doesn't hurt to leave it. Maybe the repo will get to Yarn v2/v3 someday.

Idk if it really applies to us since I think we mostly trust our collaborators but I just added it since it was a simple job https://yarnpkg.com/features/zero-installs/#does-it-have-security-implications